components/form-group/src/FormGroup.tsx

```diff
@@ -35,6 +35,9 @@ export const FormGroup: FC<FormGroupProps> = ({
   ];
   const classes = classBuilder('govuk-form-group', classBlock, classModifiers, className);
   const hintId = _hintId || `${id}-hint`;
+  const errorId = `${id}-error`;
+  const describedById = error ? errorId : hint && hintId;
```

Comment:
Looking at..: https://design-system.service.gov.uk/components/text-input/#text-input-with-errors-example-html
I don't think this fixes the bug completely. - It looks like this logic will _either_ reference the hint _or_ reference the error message. Whereas it seems that, if both are present, we should reference both.

On that note, I think we should name this constant as simply `describedBy`, as it is not necessarily a single ID.
```suggestion
  const describedBy = ([
    hint && hintId,
    error && errorId
  ]
    .filter(e => e)
    .join(' ') || undefined
  );
```
Perhaps? (There's probably a nicer way to lay that out!)

Reply:
@daniel-ac-martin Looking at the GOVUK behaviour, it does seem like they are using both when set. Your suggestion makes sense.